velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -0,0 +1,306 @@
+    Date date;
+    parseTo(string_date, date);
+    return date.days();
+  }
+
+  // For a given file at filePath, add splits equal to numSplits.
+  std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> makeSplits(
+      const std::string& filePath,
+      int64_t numSplits) const {
+    LocalReadFile lfs(filePath);
+    const int fileSize = lfs.size();
+    const int splitSize = fileSize / numSplits;
+    std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> splits;
+
+    // Add all the splits except the last one.
+    int i = 0;
+    for (; i < numSplits - 1; i++) {
+      auto split = makeHiveConnectorSplit(filePath, i * splitSize, splitSize);
+      split->fileFormat = dwio::common::FileFormat::PARQUET;
+      splits.push_back(std::move(split));
+    }
+    // Add the last split with the remaining as start.
+    auto split = makeHiveConnectorSplit(filePath, i * splitSize);
+    split->fileFormat = dwio::common::FileFormat::PARQUET;
+    splits.push_back(std::move(split));
+    return splits;
+  }
+
+  void addSplitsToTask(
+      exec::Task* task,
+      bool& noMoreSplits,
+      const std::vector<std::string>& filePaths,
+      int sourcePlanNodeId = 0,
+      int64_t numSplits = 10) const {
+    if (!noMoreSplits) {
+      for (const auto& filePath : filePaths) {
+        auto const& splits = makeSplits(filePath, numSplits);
+        for (const auto& split : splits) {
+          task->addSplit(std::to_string(sourcePlanNodeId), exec::Split(split));
+        }
+      }
+      task->noMoreSplits(std::to_string(sourcePlanNodeId));
+      noMoreSplits = true;
+    }
+  }
+
+  // Write the Lineitem TPCH table to a Parquet file and return the file
+  // location.
+  std::string writeDuckDBTPCHLineitemTableToParquet() const {
+    constexpr std::string_view tableName("lineitem");
+    // Lineitem SF=0.01 has 60175 rows
+    // Set the number of rows in a RowGroup so that the generated file contains
+    // multiple RowGroups.
+    constexpr int rowGroupNumRows = 15'000;
+    const auto& filePath =
+        fmt::format("{}/{}.parquet", tempDirectory_->path, tableName);
+    // Convert decimal columns to double.
+    const auto& query = fmt::format(
+        "COPY (SELECT l_orderkey as orderkey, l_partkey as partkey, l_suppkey as suppkey, l_linenumber as linenumber, "
+        "l_quantity::DOUBLE as quantity, l_extendedprice::DOUBLE as extendedprice, l_discount::DOUBLE as discount, "
+        "l_tax::DOUBLE as tax, l_returnflag as returnflag, l_linestatus as linestatus, "
+        "l_shipdate as shipdate, l_receiptdate as receiptdate, "
+        "l_shipinstruct as shipinstruct, l_shipmode as shipmode, l_comment as comment "
+        "FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})",
+        tableName,
+        filePath,
+        rowGroupNumRows);
+    duckDb_->execute(query);
+    return filePath;
+  }
+
+  static std::shared_ptr<DuckDbQueryRunner> duckDb_;
```

Comment:
Why can't we re-use  duckDbQueryRunner_ member variable from the base class?

Reply:
The idea is to load the TPC-H data into DuckDB only once for the entire test suite here. That is why I declared it as static and set it up inside `static void SetupTestCase`.
The base class will instantiate `DuckDbQueryRunner` for each test.

Reply:
To provide more context, I wanted to invoke `initializeTpch(kTpchScaleFactor);` only once per suite as it takes about `4 seconds` to load the data. If this is not a deal, I can use re-use the member variable.

Reply:
Thank you for explaining. It is a bit confusing to have duckDb_ and duckDbQueryRunner_ in the base class. It also means that assertQuery methods from the base class cannot be used easily. Would it not work to simply call initializeTpch() on duckDbQueryRunner_ once?